src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/GetAzureRmHubVirtualNetworkConnectionCommand.cs

```diff
@@ -0,0 +1,67 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network
+{
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Network.Models;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.Get,
+        "AzureRmHubVirtualNetworkConnection",
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSHubVirtualNetworkConnection))]
+    public class GetHubVirtualNetworkConnectionCommand : HubVnetConnectionBaseCmdlet
+    {
+        [Alias("ResourceName", "HubVirtualNetworkConnectionName")]
+        [Parameter(
+           Mandatory = false,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The parent resource name.")]
+        [ResourceGroupCompleter]
+        public string ParentResourceName { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (!string.IsNullOrWhiteSpace(this.Name))
+            {
+                WriteObject(this.GetHubVirtualNetworkConnection(this.ResourceGroupName, this.ParentResourceName, this.Name));
```

Comment:
What happens if name is provided and not parentresourcename?

Reply:
That was my bad - for child resources, we allow list only till parent. This is mainly because of performance issues and is coded the same in swagger and sdk.

ParentResourceName should be mandatory here.